src/model/ident_env.cppo.ml

```diff
@@ -129,6 +128,44 @@ let rec add_signature_type_items parent items env =
 
     | [] -> env
 
+#if OCAML_MAJOR = 4 && OCAML_MINOR >= 08
+let rec add_extended_open_items parent items env =
+  let open Types in
+    match items with
+    | Sig_type(id, _, _, _) :: rest ->
+        let env = add_extended_open_items parent rest env in
+          if Btype.is_row_name (Ident.name id) then env
+          else add_type parent id (TypeName.internal_of_ident id) env
```

Comment:
The intent was that we never want to end up referring to this type directly, and while we could do that by checking the whole path to make sure there aren't any internal names it seemed simpler to label everything underneath the parent module.

Reply:
Ah, I had understood "internal" as "the name doesn't exist in the source" rather than your "something we don't want to refer to".
Your interpretation makes sense as well.

However, the documentation on your last commit does say:
> Internal names occur when we generate items that don't explicitly exist in the source

Which isn't the case here.

Reply:
Hmm, yes, I don't think I worded that quite right - I'll fix the comment. Thanks!